src/accelerate/inference.py

```diff
@@ -59,9 +62,42 @@ def build_pipeline(model, split_points, args, kwargs) -> PipelineStage:
 def pippy_forward(forward, *args, **kwargs):
     state = PartialState()
     output = None
+
+    def _find_batch_size(args):
+        try:
+            return find_batch_size(arg)
+        except (ValueError, TypeError):
+            pass
+        return None
+
+    def _pad_inputs(args):
+        core = slice_tensors(args, slice(0, state.num_processes), process_index=0, num_processes=state.num_processes)
+        # Do args first
+        extra = slice_tensors(
+            args,
+            slice(state.num_processes, state.num_processes + 1),
+            process_index=0,
+            num_processes=state.num_processes,
+        )
+        extra = concatenate([extra] * ((found_batch_size % state.num_processes) + 1))
+        args = concatenate([core, extra])
+        return args
+
     if state.num_processes == 1:
         output = forward(*args, **kwargs)
     elif state.is_local_main_process:
+        found_batch_size = None
+        for arg in args:
+            found_batch_size = _find_batch_size(arg)
+        if found_batch_size is None:
+            for kwarg in kwargs.values():
+                found_batch_size = _find_batch_size(kwarg)
+        if found_batch_size is None:
+            raise ValueError("Could not find batch size from args or kwargs")
+        else:
+            if (found_batch_size % state.num_processes) != 0:
+                args = _pad_inputs(args)
+                kwargs = _pad_inputs(kwargs)
```

Comment:
This will work if we only have nested list/dict/tuple of tensors. Let's see if this works for most models.

Reply:
We can try it on a CV model maybe? (Since it just takes in BSxCxWxH tensors)

Reply:
It will also work on just a raw tensor :)